integration-testing/build.gradle.kts

```diff
@@ -220,6 +220,21 @@ tasks {
         classpath = sourceSet.runtimeClasspath
     }
 
+    // todo assert that this task is failing
+    // In this test coroutine debug agent is attached as a javaagent vm argument
+    // to a pure Java project (safeDebugAgentTest) with no Kotlin stdlib dependency.
+    // In this case the error should be thrown from AgetnPremain class:
+    // "java.lang.IllegalStateException: kotlinx.coroutines debug agent failed to load."
+    create<Test>("safeDebugAgentTest") {
```

Comment:
@dkhalanskyjb hi!)
I need some help with Gradle here 👀 

How could we check, that `:safeDebugAgentTest:run` fails with an error from `AgentPremain?`
Is there smth like `assertFailsWith ` in Gradle?

I've thought of an alternative to run `:safeDebugAgentTest:run` from a Kotlin test: to create a `ProcessBuilder` + try / catch, but maybe there is an easier option?

Reply:
I'm no Gradle expert, either. I've seen this approach to checking whether and how a Gradle task fails:
* https://github.com/Kotlin/kotlinx-benchmark/blob/bb2208c76f819056e1fa0e665dee503249d4650e/integration/src/test/kotlin/kotlinx/benchmark/integration/BenchmarkFunctionNameValidationTest.kt the test itself,
* https://github.com/Kotlin/kotlinx-benchmark/blob/bb2208c76f819056e1fa0e665dee503249d4650e/integration/src/test/resources/templates/funny-names the project whose build should fail,
* https://github.com/Kotlin/kotlinx-benchmark/blob/bb2208c76f819056e1fa0e665dee503249d4650e/integration/build.gradle.kts#L15 the required dependency.

I think this should point you in the right direction, but if you encounter any difficulties, please let me know, and I'll look into this more closely.

Reply:
also possibly @shanshin could help?

Reply:
I want to ensure that the `run` task of the project fails (`:safeDebugAgentTest:run`), I guess those examples from the benchmarks should help, I'll try

Thank you)